### PR TITLE
Add Whisper example for federated downstreaming

### PR DESCRIPTION
**Pull request by jafermarq**
Add a new example [`examples/whisper-federated-finetuning`](https://github.com/adap/flower/tree/whisper/examples/whisper-federated-finetuning) showing how to take a pre-trained Whisper model and federate the process of downstreaming it for the task of keyword spotting. The example works in centralised mode, simulation mode, and on raspberry pi devices.

Remaining TODOs:

- [ ] Add references to other Flower materials
- [x] Have a nice diagram at the top showing what the example is about.
- [x] Blogpost?

### CODE REVIEW

File: examples/whisper-federated-finetuning/README.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+-- sim.py            <- Runs the example with Flower simulation
+-- server.py         <- Defines the server-side logic for the on-device setting
+-- client.py         <- Defines the client-side logic for the on-device setting
+-- utils.py          <- auxiliary functions for this example
+-- centralised.py    <- Runs the example in centralized mode
+-- pyproject.toml    <- Example dependencies (if you use Poetry)
+-- requirements.txt  <- Example dependencies
+```
+
+This example can be run in different ways, please refer to the corresponding section for further instructions. This example was tested with `PyTorch 2.1.0` for all the different ways of running this example except when running on the Raspberry Pi, which seemed to only work with `PyTorch 1.13.1`. Please not the requirement files do not specify a version of PyTorch, therefore you need to choose one that works for you and your system.
```
Comment (yan-gao-GY) [inline examples/whisper-federated-finetuning/README.md:34]: A type "Please not" -> "Please note"

Comment (jafermarq) [reply examples/whisper-federated-finetuning/README.md]: thanks! fixed

**Comment by yan-gao-GY** [comment]
@jafermarq do you think it's interesting to show the memory consumption? The point may be that RPi is easy to hold whisper model and make it feasible to do real-world FL

**Comment by jafermarq** [comment]
> @jafermarq do you think it's interesting to show the memory consumption? The point may be that RPi is easy to hold whisper model and make it feasible to do real-world FL

Good point. I'll mention briefly about RAM usage. 👍

**Review by danieljanes — APPROVED** [review]

**Pull request merged**